src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,84 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { FormEvent, useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type SendDataType = {
+  newComment: CommentData,
+  setIsButtonLoading: (IsButtonLoading: boolean) => void,
+  clearText: () => void,
+};
+
+type Props = {
+  addComment: (sendData: SendDataType) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({ addComment }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [isNameEmpty, setIsNameEmpty] = useState(false);
+  const [isEmailEmpty, setIsEmailEmpty] = useState(false);
+  const [isTextEmpty, setIsTextEmpty] = useState(false);
+  const [isButtonLoading, setIsButtonLoading] = useState(false);
+
+  const clearText = () => setText('');
+
+  const onSubmitNewComment = (e: FormEvent) => {
+    e.preventDefault();
+    setIsButtonLoading(true);
+
+    if (!name) {
```

Comment:
Consider refactoring the multiple if statements in `onSubmitNewComment` to avoid setting `setIsButtonLoading(false)` multiple times. You can use a single if statement to check if any of the fields are empty 🧹